src/runtime/pay-flow.ts

```diff
@@ -586,59 +541,49 @@ export function parseSubscriptionResponse(deps, data, completeHandler) {
   );
 }
 
-/**
- * @param {!Object} swgData
- * @return {!PurchaseData}
- */
-function parsePurchaseData(swgData) {
+function parsePurchaseData(swgData: SwgCallbackData): PurchaseData {
   const raw = swgData['purchaseData'];
   const signature = swgData['purchaseDataSignature'];
   return new PurchaseData(raw, signature);
 }
 
 /**
- * @param {!Object} swgData
- * @return {?UserData}
- * @package Visible for testing.
+ * Visible for testing.
  */
-export function parseUserData(swgData) {
-  const idToken = swgData['idToken'];
+export function parseUserData(swgData: object): UserData | null {
+  const idToken = (swgData as {idToken?: string})['idToken'];
```

Comment:
Can you type `swgData` in the function signature instead, or does that break any callers?

Reply:
Good call, updated